src/main/java/startsteps/ECommerceShop/service/JwtServiceImpl.java

```diff
@@ -0,0 +1,76 @@
+package startsteps.ECommerceShop.service;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Service;
+
+import java.security.Key;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+@Service
+public class JwtServiceImpl implements JwtService {
+    @Value("${token.signing.key}")
+    private String jwtSigningKey;
+
+    @Override
+    public String extractUserName(String token) {
+
+        return extractClaim(token, Claims::getSubject);
+    }
+    private <T> T extractClaim(String token, Function<Claims, T> claimsResolvers) {
+        final Claims claims = extractAllClaims(token);
+        return claimsResolvers.apply(claims);
+    }
+
+    //private Claims extractAllClaims(String token) {
+    //    return Jwts.parser().setSigningKey(getSigningKey()).parseClaimsJws(token).getBody();
+    //}
+    private Claims extractAllClaims(String token) {
+        return Jwts.parser().setSigningKey(getSigningKey()).build().parseSignedClaims(token).getPayload();
+    }
+
+    private Key getSigningKey() {
+        byte[] keyBytes = Decoders.BASE64.decode(jwtSigningKey);
+        return Keys.hmacShaKeyFor(keyBytes);
+    }
+
+    @Override
+    public String generateToken(UserDetails userDetails) {
+        return generateToken(new HashMap<>(),userDetails);
+    }
+
+    private String generateToken(Map<String, Object> extraClaims, UserDetails userDetails) {
+        Date now = new Date();
+        Date expirationDate = new Date(now.getTime() + 1000 * 60 * 24);
```

Comment:
is this 24 minutes or 24 hours?

Reply:
it this case 24minutes, it is my mistake, i would like to do 24 hours.
I have already change it for 24 hours